nestkernel/node_collection.h

```diff
@@ -384,6 +379,16 @@ class NodeCollectionPrimitive : public NodeCollection
   index last_;                         //!< The last node ID in the primitive
   index model_id_;                     //!< Model ID of the node IDs
   NodeCollectionMetadataPTR metadata_; //!< Pointer to the metadata of the node IDs
+  bool nodes_have_no_proxies_;         //!< Whether the primitive contains devices or not
+
+  /**
+   * Returns whether the model ids of all nodes in the primitive are the same as the expected model id.
+   *
+   * @param model_id Expected model id.
+   * @return true if model ids of all nodes in the primitive are the same as the expected model id, false otherwise.
+   *
+   */
+  bool all_model_ids_same_as_expected_( index ) const;
```

Comment:
This seems confusing. How could it ever be false for a primitive and thus homogeneous node collection?